pkg/controller/job/indexed_job_utils.go

```diff
@@ -34,12 +34,13 @@ const (
 	unknownCompletionIndex = -1
 )
 
-// calculateCompletedIndexesStr returns a string representation of the list
-// of completed indexes in compressed format.
-func calculateCompletedIndexesStr(pods []*v1.Pod) string {
+// calculateCompletedIndexes returns a string representation of the list of
+// completed indexes in compressed format and the number of completed indexes.
+func calculateCompletedIndexes(pods []*v1.Pod) (string, int32) {
```

Comment:
is it completed or succeeded (completed could mean a failed pod)

Reply:
There is already some mixed messages in the API: a Pod can be Failed or Succeeded.
But a Job can be Failed or Complete.

In that sense, complete and succeeded are synonyms. At no point failed is considered completed.